apparelo/apparelo/doctype/lot_creation/custom_scripts.py

```diff
@@ -8,19 +8,22 @@
 
 
 def create_surplus_location_warehouse(doc, action):
-    if action == 'validate':
-        if not frappe.db.exists("Warehouse", f"Surplus-{doc.name} - {abbr}"):
-            frappe.get_doc({"doctype": "Warehouse", "warehouse_name": f"Surplus - {doc.name}",
-                            "is_group": 0, "parent_warehouse": f"Surplus Warehouse - {abbr}"}).save()
-        if not frappe.db.exists("Warehouse", f"Surplus Mistake - {doc.name} - {abbr}"):
-            frappe.get_doc({"doctype": "Warehouse", "warehouse_name": f"Surplus Mistake - {doc.name}",
-                            "is_group": 0, "parent_warehouse": f"Surplus Warehouse - {abbr}"}).save()
+	if action == 'after_insert':
+		surplus_location_warehouse=frappe.db.get_value("Warehouse",{"parent_warehouse":"Surplus","location":doc.name},"name")
+		surplus_mistake_warehouse=frappe.db.get_value("Warehouse",{"parent_warehouse":"Surplus","location":doc.name,"warehouse_type":"Mistake"},"name")
```

Comment:
What is the location mentioned here?

Reply:
Location field in warehouse.Whenever the user creates new location, the surplus warehouse is created with the location field and it helps to filter the warehouse based on the location field.